src/mame/drivers/skyfox.cpp

```diff
@@ -55,7 +55,7 @@ void skyfox_state::skyfox_map(address_map &map)
 	map(0x0000, 0xbfff).rom();
 	map(0xc000, 0xcfff).ram();
 	map(0xd000, 0xd3ff).ram().share("spriteram");
-	map(0xd400, 0xdfff).ram(); // ?
+	map(0xd400, 0xdfff).ram().share("bgram"); // Probably for background
```

Comment:
Probably?

Reply:
Certainly it contains stars position, but and also contains other unknown values.

Reply:
Generally a spriteram and I guess this bgram is just a RAM buffer with nothing special attached. 
It's generally a video chip(s) responsibility to actually copy the blocks from RAM to its internal buffer before working on it, and being a DMA operation it can have arbitrary boundaries while not touching the rest. I'm gonna bet the unused values are just used by the SW for storing internal state values.

Reply:
The subroutine that update stars is at $7330 (in Japan version), but it looks like writing to $d4e0-$d4ff only.
Also, this looks like part of vram because memory map is continuous with sprite area.

Reply:
$d400-$d4ff is background vram related, the first $e0 area has hardcoded values which needs to be tested if they change later in RAM.
$de00-$dfff is definitely scratch RAM (Stack Pointer is there)
The stuff in the middle suspiciously look tables to me, and a couple gets read back. 
Given the Jaleco love for mirroring may even be that A11 is unconnected, you can give a shot with a `.mirror(0x800)` on the $dxxx area map. 
In general driver needs some love, `skyfox_vregs_w` should actually be `output_w` but it's not from this PR anyway so it can be safely merged as-is imho.

Reply:
Thank you. It was helpful.
I fixed the following :
- Divide background VRAM ($d400-$d4ff) and work RAM ($d500-$dfff) in address map
- Rename `skyfox_vregs_w` to `output_w`
- Remove undefined interrupt declaration from header